### PR TITLE
LSM-6 ~ Skip avro header for signature.

### PR DESCRIPTION
**Pull request by kevinsmeets**

### CODE REVIEW

File: osgp/shared/osgp-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/signature/MessageSigner.java [hunk 1]
```diff
@@ -56,8 +57,13 @@
     public static final String DEFAULT_SIGNATURE_KEY_ALGORITHM = "RSA";
     public static final int DEFAULT_SIGNATURE_KEY_SIZE = 2048;
 
+    // Two magic bytes (0xC3, 0x01) followed by an 8-byte fingerprint
+    public static final int AVRO_HEADER_LENGTH = 10;
+
     private boolean isSigningEnabled;
 
+    private boolean isStripHeaders;
```
Comment (smvdheijden) [inline osgp/shared/osgp-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/signature/MessageSigner.java:65]: I thought that our convention was to use the `is` prefix only for getters, not for fields?

File: osgp/shared/osgp-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/signature/MessageSigner.java [hunk 2]
```diff
@@ -210,6 +229,19 @@ public boolean verify(final Message message) {
         }
     }
 
+    private boolean hasAvroHeader(final byte[] bytes) {
+        return bytes.length >= AVRO_HEADER_LENGTH && (bytes[0] & 0xFF) == 0xC3 && (bytes[1] & 0xFF) == 0x01;
+    }
+
+    private byte[] toSchemalessBytes(final ByteBuffer byteBuffer) {
```
Comment (smvdheijden) [inline osgp/shared/osgp-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/signature/MessageSigner.java:236]: consider renaming to toHeaderlessBytes.

File: osgp/shared/osgp-kafka-distributionautomation/src/test/java/org/opensmartgridplatform/adapter/kafka/da/signature/MessageSignerTest.java [hunk 3]
```diff
@@ -34,6 +34,8 @@
 
     private static final boolean IS_SIGNING_ENABLED = true;
 
+    private static final boolean IS_STRIP_HEADERS = true;
```
Comment (smvdheijden) [inline osgp/shared/osgp-kafka-distributionautomation/src/test/java/org/opensmartgridplatform/adapter/kafka/da/signature/MessageSignerTest.java:37]: I thought that our convention was to use the `is` prefix only for getters, not for fields?

File: osgp/shared/osgp-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/signature/MessageSigner.java [hunk 4]
```diff
@@ -345,6 +381,8 @@ public static Builder newBuilder() {
 
         private boolean isSigningEnabled;
 
+        private boolean isStripHeaders;
```
Comment (smvdheijden) [inline osgp/shared/osgp-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/signature/MessageSigner.java:384]: I thought that our convention was to use the is prefix only for getters, not for fields?

**Review by smvdheijden — CHANGES_REQUESTED** [review]

**Review by smvdheijden — APPROVED** [review]

**Comment by kevinsmeets** [comment]
retest this please

**Pull request merged**